### PR TITLE
Fixes multi inline select menu issues

### LINKED ISSUE #7918: Selectmenu - mini, inline, multiple - count indicator visual glitch
Demo: http://jsbin.com/bupisimofo/1/edit?html,css,output
Count indicator position must be different when mini="true".

### LINKED ISSUE #8038: Multi inline select menu hides if width exceeds device width
The width of the multi inline select menu keeps increasing on selecting new options and finally the extra width gets hidden. This issue is only appeared when data-inline="true" is set.
test page:  http://jsbin.com/yobacaveqo

Steps to reproduce
<ol>
<li>Visit this issue template provided above.</li>
<li>Add new options by selecting them</li>
</ol>

### PR DESCRIPTION
**Pull request by kapilgarg1996**
Fixes #7918
Fixes #8038

### CODE REVIEW

File: css/structure/jquery.mobile.core.css [hunk 1]
```diff
@@ -279,6 +279,7 @@ div.ui-mobile-viewport {
 	padding: 0;
 }
 .ui-btn-inline {
+	max-width:100% ;
```
Comment (arschmitz) [inline css/structure/jquery.mobile.core.css]: I don't think this should be part of the inline rule instead the `ui-select` rule I don't think we should restrict the width of inline buttons in general.  @jaspermdegroot @sfrisk thoughts?

Comment (kapilgarg1996) [inline css/structure/jquery.mobile.core.css]: @arschmitz  Firstly i was thinking of adding a new class to selectmenu which restricts its width when it goes out of the page. This class would only be added to element having .ui-select class and only when it goes out of the page. Do you think this idea is better..??

File: css/structure/jquery.mobile.forms.select.css [hunk 2]
```diff
@@ -66,6 +66,10 @@
 .ui-select .ui-btn-icon-right .ui-li-count {
 	right: 3.2em;
 }
+/*Fixes #7918 : Count indicator position must be different for mini version*/
+.ui-select.ui-mini .ui-btn-icon-right .ui-li-count {
+  right: 2.6em;
```
Comment (sfrisk) [inline css/structure/jquery.mobile.forms.select.css]: Indent using tabs, not spaces:
http://contribute.jquery.org/style-guide/css/#spacing

File: css/structure/jquery.mobile.core.css (hunk 1 above)
Comment (jaspermdegroot) [inline css/structure/jquery.mobile.core.css]: @arschmitz 

Agreed. Let's fix this in the selectmenu CSS.

I think we should also look at old (closed) tickets because I know we have been dealing with select width issues in the past and don't want regressions. Might be good to add a unit test for this.

We should do some testing with ```max-width: 100%;``` for the selectmenu button in the ui-1-12 branch after my CSS update (no PR yet) landed in there.

Comment (kapilgarg1996) [inline css/structure/jquery.mobile.core.css]: @jaspermdegroot test wriiten :+1: 
please check if its correct

Comment (kapilgarg1996) [inline css/structure/jquery.mobile.core.css]: @jaspermdegroot done :+1: 
unit test written..please check if its correct..

File: tests/unit/select/select_core.js [hunk 3]
```diff
@@ -42,6 +42,30 @@
 			"Checkboxes should not have ui-checkbox-on class" );
 	});
 
+	module( "Custom select Multiple Inline" ) ;
+
+	test( "Custom select Multiple Inline width is adjusted correctly", function() {
+	var selectMenu = $( "#width-test" );
+	var parent = $( "#width-test" ).parent();
+	var widths = [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000];
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: missing spaces in side brackets

File: tests/unit/select/select_core.js [hunk 4]
```diff
@@ -42,6 +42,30 @@
...
+	var finalResult = 0 ;
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: blank line after var statements

File: tests/unit/select/select_core.js [hunk 5]
```diff
@@ -42,6 +42,30 @@
...
+	$.each(widths , function(index, width){
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: missing spaces inside parens

File: tests/unit/select/select_core.js [hunk 6]
```diff
@@ -42,6 +42,30 @@
...
+		parent.css({"width": width+"px"}) ;
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: missing spaces

File: tests/unit/select/select_core.js [hunk 7]
```diff
@@ -42,6 +42,30 @@
...
+		$( "#width-test" )
+			.find( "option" )
+				.attr( "selected", true )
+				.prop( "selected", true )
+			.end()
+			.selectmenu( "refresh" );
+	var result = ( selectMenu.width() < parent.width() ? 1 : 0 );
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: declare all variables at the top of the function

File: tests/unit/select/select_core.js [hunk 8]
```diff
@@ -42,6 +42,30 @@
...
+	if(result == 1)
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: spaces

File: tests/unit/select/select_core.js [hunk 9]
```diff
@@ -42,6 +42,30 @@
...
+	finalResult += 1 ;
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: indententation

File: tests/unit/select/select_core.js [hunk 10]
```diff
@@ -42,6 +42,30 @@
...
+	}) ;
+			deepEqual( finalResult, 10,
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: `deepEqual` is only for arrays, objects, regular expressions, dates and functions. http://api.qunitjs.com/deepEqual/

File: tests/unit/select/select_core.js [hunk 11]
```diff
@@ -42,6 +42,30 @@
...
+	var selectMenu = $( "#width-test" );
+	var parent = $( "#width-test" ).parent();
+	var widths = [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000];
+	var finalResult = 0 ;
+	$.each(widths , function(index, width){
+		parent.css({"width": width+"px"}) ;
+		$( "#width-test" )
+			.find( "option" )
+				.attr( "selected", true )
+				.prop( "selected", true )
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: you don't need to set both the prop and that ater you should only be setting the prop

Comment (arschmitz) [inline tests/unit/select/select_core.js]: your setting ALL the options to selected here

File: tests/unit/select/select_core.js (hunk 7 above)
Comment (arschmitz) [inline tests/unit/select/select_core.js]: no need for parens here

File: tests/unit/select/select_core.js [hunk 12]
```diff
@@ -42,6 +42,30 @@
...
+			.end()
+			.selectmenu( "refresh" );
+	var result = ( selectMenu.width() < parent.width() ? 1 : 0 );
+	if(result == 1)
+	finalResult += 1 ;
+	}) ;
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: missing space

File: tests/unit/select/select_core.js (hunk 8 above)
Comment (arschmitz) [inline tests/unit/select/select_core.js]: always use strict comparison with `===`

File: tests/unit/select/select_core.js [hunk 13]
```diff
@@ -42,6 +42,30 @@
...
+			deepEqual( finalResult, 10,
+			"select menu width should not exceed parent's width" );
+	}) ;
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: missing space

File: tests/unit/select/index.html [hunk 14]
```diff
@@ -43,6 +43,25 @@
 			<option value="4">The 4th Option</option>
 			</select>
 		</div>
+		<div id="width-test-container">
+			<div data-role="fieldcontain">
+				<label for="width-test">Multiple:</label>
+				<select name="width-test" id="width-test" multiple="multiple" data-inline="true" data-nstest-native-menu="false" data-nstest-overlay-theme="b" data-nstest-theme="x">
```
Comment (arschmitz) [inline tests/unit/select/index.html]: `multiple` is a property just need `multiple` not `multiple=multiple`

File: tests/unit/select/select_core.js (hunk 3 above)
Comment (arschmitz) [inline tests/unit/select/select_core.js]: testing this many widths seems like overkill

File: tests/unit/select/select_core.js [hunk 15]
```diff
@@ -42,6 +42,30 @@
 			"Checkboxes should not have ui-checkbox-on class" );
 	});
 
+	module( "Custom select Multiple Inline" ) ;
+
+	test( "Custom select Multiple Inline width is adjusted correctly", function() {
+	var selectMenu = $( "#width-test" );
+	var parent = $( "#width-test" ).parent();
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: do `selectMenu.parent()`

Comment (kapilgarg1996) [inline tests/unit/select/select_core.js]: Thanks for the review.
I'll update the p.r. as you instructed

File: tests/unit/select/select_core.js (hunk 11 above)
Comment (kapilgarg1996) [inline tests/unit/select/select_core.js]: setting all the options selected ensures that even if all the options are selected, the selectmenu width isn't exceeding parent's width.
Isn't it okay ?

File: tests/unit/select/index.html (hunk 14 above)
Comment (kapilgarg1996) [inline tests/unit/select/index.html]: Oh alright..
Then it must also implemented in the demos too..
http://demos.jquerymobile.com/1.4.5/selectmenu-custom/#select-custom-19

File: css/structure/jquery.mobile.forms.select.css [hunk 16]
```diff
@@ -66,6 +66,10 @@
 .ui-select .ui-btn-icon-right .ui-li-count {
 	right: 3.2em;
 }
+/*Fixes #7918 : Count indicator position must be different for mini version*/
```
Comment (arschmitz) [inline css/structure/jquery.mobile.forms.select.css]: Don't reference issues in comments. Blank line before comments. Space before first word.

File: css/structure/jquery.mobile.forms.select.css [hunk 17]
```diff
@@ -135,3 +139,7 @@
 .ui-selectmenu .ui-selectmenu-placeholder {
 	display: none;
 }
+/*Fixes #8038: Multi inline select menu hides if width exceeds device width*/
```
Comment (arschmitz) [inline css/structure/jquery.mobile.forms.select.css]: Same as on comment above

File: tests/unit/select/index.html (hunk 14 above)
Comment (arschmitz) [inline tests/unit/select/index.html]: yes any where we use it this should be fixed. that would be a separate PR though.

File: tests/unit/select/select_core.js [hunk 18]
```diff
@@ -42,6 +42,30 @@
...
+	test( "Custom select multiple inline width is adjusted correctly", function() {
+		var selectMenu = $( "#width-test" ),
+			parent = selectMenu.parent(),
+			widths = [ 100, 250, 500, 1000 ],
+			finalResult = 0,
+			result = 0;
+
+		$.each( widths , function( index, width ){
+			parent.css( {"width": width+"px"} );
+			$( "#width-test" )
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: use `selectMenu`

File: tests/unit/select/select_core.js [hunk 19]
```diff
@@ -42,6 +42,30 @@
...
+				.find( "option" )
+				.prop( "selected", true )
+				.end()
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: Remove end and end chain after prop then do `selectMenu.selectmenu( "refresh" );`
It will be much more clear what your doing

File: tests/unit/select/select_core.js [hunk 20]
```diff
@@ -42,6 +42,30 @@
...
+				.selectmenu( "refresh" );
+			result = selectMenu.width() < parent.width() ? 1 : 0;
+			if( result === 1 )
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: space before opening paren, always use brackets with if statements.

File: tests/unit/select/select_core.js [hunk 21]
```diff
@@ -42,6 +42,30 @@
...
+
+	test( "Custom select multiple inline width is adjusted correctly", function() {
+		var selectMenu = $( "#width-test" ),
+			parent = selectMenu.parent(),
+			widths = [ 100, 250, 500, 1000 ],
+			finalResult = 0,
+			result = 0;
+
+		$.each( widths , function( index, width ){
+			parent.css( {"width": width+"px"} );
```
Comment (arschmitz) [inline tests/unit/select/select_core.js]: spaces inside brackets
